### PR TITLE
Add transaction error handling

### PR DESCRIPTION
**Pull request by pivilartisant**
Add error handling on transaction and on sign

### CODE REVIEW

File: internal/handler/wallet/sign.go [hunk 1]
```diff
@@ -109,6 +110,11 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
 
 	signature, err := wlt.Sign(signingOperation, op)
 	if err != nil {
+
+		// event emit false
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:114]:
```suggestion
```

File: internal/handler/wallet/sign.go [hunk 2]
```diff
@@ -125,6 +131,10 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
 }
 
 func (s *walletSign) handleBadRequest(errorCode string) middleware.Responder {
+	// event emit false
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:134]:
```suggestion
```

File: internal/handler/wallet/sign.go
Comment (Thykof) [inline internal/handler/wallet/sign.go:1]: we need to see for proper code message

File: wails-frontend/src/pages/PasswordPromptHandler/Default.tsx [hunk 3]
```diff
@@ -82,7 +86,7 @@ export function Default() {
             defaultValue=""
             name="password"
             placeholder="Password"
-            error={error?.password}
+            error={errorMessage ?? error?.password}
```
Comment (Thykof) [inline wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: does the UI render pretty?  I would expect a separated `<p>` bellow.

**Review by Thykof — CHANGES_REQUESTED** [review]

File: wails-frontend/src/pages/PasswordPromptHandler/Default.tsx (hunk 3 above)
Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: ui does propoerly render errorMessage if not null or fallsback to error.password

Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: Screenshot 2023-08-25 at 14 29 01

test method : ``` if (!Success) {
      if (CodeMessage === ErrorCode.WrongPassword) {
        setErrorMessage('Failed to process transaction');
        return;
      } else {
        setError({ password: Intl.t(`errors.${CodeMessage}`) });
        return;
      }
    }```

Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: but i guess most optimal would be to just reuse the errorObject we already have, that way we can remove the errorMessage useState, wdyt @Thykof ?

File: internal/handler/wallet/sign.go
Comment (pivilartisant) [reply internal/handler/wallet/sign.go:1]: let me know if you want to sync about itn or just keep me update on necessary changes

File: wails-frontend/src/pages/PasswordPromptHandler/Default.tsx (hunk 3 above)
Comment (Thykof) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: we should not mix errors related to a field of the form and error related to the submit action

Comment (Thykof) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: see for example here
https://github.com/massalabs/station-massa-wallet/blob/6c317bd32a8608bf55ce9d772acf4092931b0e83/wails-frontend/src/pages/backupMethods.tsx#L49

we have an independent local state and html tag

Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: true but in your example, the error does not live in a component; here we have the input component which wraps automatically the value of ```error``` in a div tag -> https://github.com/massalabs/ui-kit/blob/main/src/components/Input/Input.tsx

Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: understood ! will keep error render separate !

Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:89]: Screenshot 2023-08-25 at 15 42 20

File: wails-frontend/src/pages/PasswordPromptHandler/Default.tsx [hunk 4]
```diff
@@ -56,6 +57,9 @@ export function Default() {
       if (CodeMessage === ErrorCode.WrongPassword) {
         setError({ password: Intl.t(`errors.${CodeMessage}`) });
         return;
+      } else {
+        setErrorMessage(Intl.t(`errors.default`));
+        return;
       }
```
Comment (Thykof) [inline wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:63]: you chose ;) I prefer the first one bellow:

```
      if (CodeMessage === ErrorCode.WrongPassword) {
        setError({ password: Intl.t(`errors.${CodeMessage}`) });
      } else {
        setErrorMessage(Intl.t(`errors.default`));
      }
      return
```

```
      if (CodeMessage === ErrorCode.WrongPassword) {
        setError({ password: Intl.t(`errors.${CodeMessage}`) });
        return
      }
      setErrorMessage(Intl.t(`errors.default`));
      return
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: internal/handler/wallet/sign.go [hunk 5]
```diff
@@ -109,6 +110,10 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
 
 	signature, err := wlt.Sign(signingOperation, op)
 	if err != nil {
+
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply internal/handler/wallet/sign.go:113]: Screenshot 2023-08-25 at 15 42 20
it's not picking up on this but i'll got in manually

File: wails-frontend/src/pages/PasswordPromptHandler/Default.tsx (hunk 4 above)
Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:63]: the first one it is !

File: internal/handler/wallet/sign.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Thykof) [reply internal/handler/wallet/sign.go:113]: ok

File: wails-frontend/src/pages/PasswordPromptHandler/Default.tsx [hunk 6]
```diff
@@ -56,7 +57,10 @@ export function Default() {
       if (CodeMessage === ErrorCode.WrongPassword) {
         setError({ password: Intl.t(`errors.${CodeMessage}`) });
         return;
```
Comment (Thykof) [inline wails-frontend/src/pages/PasswordPromptHandler/Default.tsx:59]:
```suggestion
```

File: wails-frontend/src/pages/PasswordPromptHandler/Delete.tsx [hunk 7]
```diff
@@ -44,7 +45,10 @@ export function Delete() {
       if (CodeMessage === ErrorCode.WrongPassword) {
         setError({ password: Intl.t(`errors.${CodeMessage}`) });
         return;
```
Comment (Thykof) [inline wails-frontend/src/pages/PasswordPromptHandler/Delete.tsx:47]:
```suggestion
```

File: wails-frontend/src/pages/PasswordPromptHandler/Sign.tsx [hunk 8]
```diff
@@ -60,7 +61,10 @@ export function Sign() {
       if (CodeMessage === ErrorCode.WrongPassword) {
         setError({ password: Intl.t(`errors.${CodeMessage}`) });
         return;
```
Comment (Thykof) [inline wails-frontend/src/pages/PasswordPromptHandler/Sign.tsx:63]:
```suggestion
```

File: wails-frontend/src/pages/PasswordPromptHandler/Transfer.tsx [hunk 9]
```diff
@@ -52,7 +53,10 @@ export function Transfer() {
       if (CodeMessage === ErrorCode.WrongPassword) {
         setError({ password: Intl.t(`errors.${CodeMessage}`) });
         return;
```
Comment (Thykof) [inline wails-frontend/src/pages/PasswordPromptHandler/Transfer.tsx:55]:
```suggestion
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: wails-frontend/src/pages/PasswordPromptHandler/Transfer.tsx (hunk 9 above)
Comment (pivilartisant) [reply wails-frontend/src/pages/PasswordPromptHandler/Transfer.tsx]: sorry i was careful enough ^^

File: internal/handler/wallet/sign.go
Comment (Thykof) [reply internal/handler/wallet/sign.go:1]: we have:
- errorSignDecodeOperation: "Wallet-0011", it means that the provided operation is not base64 valid,
- errorSignDecodeMessage: "Wallet-0012", it means that the operation is not valid

Comment (Thykof) [reply internal/handler/wallet/sign.go:1]: I pushed two commit for error codes, you can now translate them with @LucasF243 and @longgamma-massa

Comment (LucasF243) [reply internal/handler/wallet/sign.go:1]: If the user can't remedy the error, he doesn't need to know the precise details of the error: "failed to signoperation" is sufficient

Comment (longgamma-massa) [reply internal/handler/wallet/sign.go:1]: 1. FOR errorSignDecodeOperation: "Wallet-0011", SHOW: Failed to sign the operation (not a valid base 64 string)

@peterjah @pivilartisant  I need confirmation for this message (did I properly say it's a string)?
2. FOR errorSignDecodeMessage: "Wallet-0012", SHOW: Failed to sign the operation (invalid operation)

Comment (pivilartisant) [reply internal/handler/wallet/sign.go:1]:
```
"Wallet-0011": "Failed to sign: operation is not base 64 valid",
"Wallet-0012": "Failed to sign: invalid operation",
```
@longgamma-massa

Comment (longgamma-massa) [reply internal/handler/wallet/sign.go:1]: go @pivilartisant

**Review by Thykof — APPROVED** [review]

**Review by peterjah — APPROVED** [review]

**Pull request merged**